subsys/shell/modules/kernel_service.c

```diff
@@ -218,6 +222,55 @@ static int cmd_kernel_stacks(const struct shell *shell,
 }
 #endif
 
+static int cmd_kernel_sleep(const struct shell *sh,
+			    size_t argc, char **argv)
+{
+	ARG_UNUSED(sh);
+	ARG_UNUSED(argc);
+
+	k_msleep(strtoul(argv[1], NULL, 10));
+
+	return 0;
+}
+
+#if defined(CONFIG_LOG_RUNTIME_FILTERING)
+static int16_t log_source_id_get(const char *name)
+{
+	int16_t count = (int16_t)log_src_cnt_get(CONFIG_LOG_DOMAIN_ID);
+
+	for (int16_t i = 0; i < count; i++) {
+		if (strcmp(log_source_name_get(CONFIG_LOG_DOMAIN_ID, i), name)
+		    == 0) {
+			return i;
+		}
+	}
+	return -1;
+}
```

Comment:
I'm confused about the use of `int16_t` here; why does `log_src_cnt_get` return a uint32_t, but we use a (much) smaller data type (`int16_t`) here?

Reply:
good point. there's a very simple reason for that though: mindless [copy-and-paste](https://github.com/jori-nordic/zephyr/blob/edb60e507b446c6d0d1595b0e77dc48f05f1435a/tests/subsys/logging/log_api/src/main.c#L70) . 
I think the real reason is that `log_filter_set` expects an int16_t. I added a check and reworked the fn a little bit.